### PR TITLE
Aborts stream file reader once metadata are returned

### PR DESCRIPTION
**Pull request by rodu**
- Returns the file reader from wrapper to allow abort method call

This change relates to issue 115: https://github.com/leetreveil/musicmetadata/issues/115

### CODE REVIEW

**Comment by leetreveil** [comment]
I've fixed the test runner issue. Can you rebase?

**Comment by rodu** [comment]
Sure, just rebased.

File: lib/browser.js [hunk 1]
```diff
@@ -6,11 +6,25 @@ var musicmetadata = require('./index')
 var Stream = require('stream').Stream
 
 module.exports = function (stream, opts, callback) {
-  return musicmetadata(wrapFileWithStream(stream), opts, callback)
+  var wrappedStram = wrapFileWithStream(stream)
```
Comment (leetreveil) [inline lib/browser.js]: should be `wrappedStream`.

File: lib/browser.js [hunk 2]
```diff
@@ -34,7 +48,10 @@ function wrapFileWithStream (file) {
     throw new Error('You must provide a valid File or Blob object')
   }
 
-  return readStream(file).pipe(stream)
```
Comment (leetreveil) [inline lib/browser.js]: Can we revert this back to a one liner: `return readStream(file).pipe(stream)`

**Comment by leetreveil** [comment]
I wonder if a better solution would be to add an option to allow `musicmetadata` to auto close the underlying stream once parsing has finished?

Something like:
```javascript
var mm = musicmetadata(stream, {autoClose: true})
```

File: lib/browser.js (hunk 2 above)
Comment (rodu) [inline lib/browser.js]: In order to call the `abort()` method you need to return the FileStream instance. The `pipe()` method returns the destination only, that's why I had to change it not to be in a single line.
See: node_modules/filereader-stream/index.js#75

**Comment by Skeen** [comment]
This patch seems to have broken my code. My input is a stream, and it does NOT have a '.dest' field.
- Additionally I do not believe it to be the general interface of node streams to have dest fields.

**Pull request merged**